### ISSUE TITLE
Delete deprecated optparse code

### ISSUE THREAD

**Issue by tleonhardt** [python-cmd2/cmd2#340]
For the 0.8.0 release of ``cmd2`` on February 1, 2018 we officially deprecated all usage of the **optparse** module within ``cmd2`` and replaced it with **argparse**-based decorators.

We should plan to delete the deprecated **optparse** code in either the 0.8.3 or 0.8.4 release.

Any opinions regarding which of these releases it should be deprecated in?

**Comment by tleonhardt** [comment]
Ok, unless there are any objections, lets target deleting the old **optparse** code for the 0.8.4 release.

**Comment by kotfu** [comment]
No objections here.

**Comment by tleonhardt** [comment]
@kmvanbrunt Do you anticipate any negative end user impact if we delete the **optparse** code in the next release - i.e. will you be able to port all applications to using the **argparse** decorators?

**Comment by kmvanbrunt** [comment]
We started that conversion today. I don't anticipate another release for
months, so its fine to delete the code now.

**Comment by kmvanbrunt** [comment]
No, we have plenty of time to port the old **optparse** code to **argparse**
